code/scripts/espurna_utils/bsec.py

```diff
@@ -0,0 +1,58 @@
+import os
+from SCons.Script import Move, Delete, Mkdir, Touch
+
+
+def bsec_inject_patcher(env):
+    bsec_builder = next(
+        (
+            builder
+            for builder in env.GetLibBuilders()
+            if builder.name == "BSEC Software Library"
+        ),
+        None,
+    )
+
+    if bsec_builder is None:
+        return
+
+    bsec_dir = os.path.join(bsec_builder.src_dir, "esp8266")
+
+    out_path = os.path.join(bsec_dir, "out")
+    patchflag_path = os.path.join(bsec_dir, ".patched")
+
+    # Note:
+    # depending on libalgobsec.a or libalgobsec.a.orig will force SCons to always rebuild
+    # this happens b/c last time it had seen it and calculated deps, there was still the original .a
+    patcher = env.Command(
+        patchflag_path,
+        None,
+        [
+            Move("${TARGET.dir}/libalgobsec.a.orig", "${TARGET.dir}/libalgobsec.a"),
+            Mkdir(out_path),
+            "cd ${TARGET.dir}/out && $AR vx ../libalgobsec.a.orig",
+            rename_o_to_c_o_files("out"),
+            "cd ${TARGET.dir} && $AR vcr libalgobsec.a out/*.c.o",
+            Delete(out_path),
+            Touch(patchflag_path),
+        ],
+    )
+
+    action = env.VerboseAction(patcher, "Patching libalgobsec.a")
```

Comment:
idk if this actually does anything... actions underneath aren't hidden when not verbose

Reply:
`patcher = env.Alias("patch-bsec", patchflag_path)` seems to be enough here. Shall I remove it?

Reply:
Ok